scripts/preprocess_waymo.py

```diff
@@ -0,0 +1,450 @@
+
+        self.lidar_list = [
+            '_FRONT', '_FRONT_RIGHT', '_FRONT_LEFT', '_SIDE_RIGHT', '_SIDE_LEFT'
+        ]
+        self.type_list = ['UNKNOWN', 'VEHICLE', 'PEDESTRIAN', 'SIGN', 'CYCLIST']
+
+        self.selected_waymo_classes = self.classes
+
+        self.selected_waymo_locations = None
+
+        self.dataset_path = dataset_path
+        self.save_dir = save_dir
+        self.workers = int(workers)
+        self.is_test = is_test
+        self.prefix = ''
+        self.save_track_id = False
+
+        self.tfrecord_files = sorted(
+            glob.glob(join(self.dataset_path, "*.tfrecord")))
+
+        self.label_save_dir = f'{self.save_dir}/label_'
+        self.label_all_save_dir = f'{self.save_dir}/label_all'
+        self.image_save_dir = f'{self.save_dir}/image_'
+        self.calib_save_dir = f'{self.save_dir}/calib'
+        self.point_cloud_save_dir = f'{self.save_dir}/velodyne'
+        self.pose_save_dir = f'{self.save_dir}/pose'
+
+        self.create_folder()
+
+    def create_folder(self):
+        if not self.is_test:
+            dir_list1 = [
+                self.label_all_save_dir, self.calib_save_dir,
+                self.point_cloud_save_dir, self.pose_save_dir
+            ]
+            dir_list2 = [self.label_save_dir, self.image_save_dir]
+        else:
+            dir_list1 = [
+                self.calib_save_dir, self.point_cloud_save_dir,
+                self.pose_save_dir
+            ]
+            dir_list2 = [self.image_save_dir]
+        for d in dir_list1:
+            makedirs(d, exist_ok=True)
+        for d in dir_list2:
+            for i in range(5):
+                makedirs(f'{d}{str(i)}', exist_ok=True)
+
+    def convert(self):
+        print("Start converting ...")
+        with Pool(self.workers) as p:
+            p.map(self.process_one, [i for i in range(len(self))])
+
+    def process_one(self, file_idx):
+        path = self.tfrecord_files[file_idx]
+        dataset = tf.data.TFRecordDataset(path, compression_type='')
+
+        for frame_idx, data in enumerate(dataset):
+            frame = dataset_pb2.Frame()
+            frame.ParseFromString(bytearray(data.numpy()))
+
+            if (self.selected_waymo_locations is not None and
+                    frame.context.stats.location
+                    not in self.selected_waymo_locations):
+                print("continue")
+                continue
+
+            if self.write_image:
+                self.save_image(frame, file_idx, frame_idx)
+            self.save_calib(frame, file_idx, frame_idx)
+            self.save_lidar(frame, file_idx, frame_idx)
+            self.save_pose(frame, file_idx, frame_idx)
+
+            if not self.is_test:
+                self.save_label(frame, file_idx, frame_idx)
+
+    def __len__(self):
+        return len(self.tfrecord_files)
+
+    def save_image(self, frame, file_idx, frame_idx):
+        self.prefix = ''
+        import cv2
```

Comment:
Is it necessary to add `cv2` as a dependency in our package?  If we only want to use imwrite.

Reply:
Done. changed to npy files